Bug report: Unable to Interpolate Models

**Information:**
```
Error

An error occurred in a PyTorch Interpolate Models node:

Input and output sizes should be greater than 0, but got input (H: 1, W: 1) output (H: 0, W: 0)
```
ChaiNNer: chaiNNer 0.15.2
OS: macOS 10.18

**Description**
When attempting to interpolate models the above error is thrown.


**Logs**
```
[2022-11-24 14:56:57.864] [info]  Backend: [67906] [ERROR] Traceback (most recent call last):
  File "/Applications/chaiNNer.app/Contents/Resources/src/process.py", line 344, in __process
    raw_output, execution_time = await self.loop.run_in_executor(
  File "/Users/imt/Library/Application Support/chaiNNer/python/python/lib/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/Applications/chaiNNer.app/Contents/Resources/src/process.py", line 222, in wrapper
    result = supplier()
  File "/Applications/chaiNNer.app/Contents/Resources/src/nodes/nodes/pytorch/interpolate_models.py", line 101, in run
    if not self.check_can_interp(state_a, state_b):
  File "/Applications/chaiNNer.app/Contents/Resources/src/nodes/nodes/pytorch/interpolate_models.py", line 81, in check_can_interp
    t_out = model(img_tensor)
  File "/Users/imt/Library/Application Support/chaiNNer/python/python/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/Applications/chaiNNer.app/Contents/Resources/src/nodes/utils/architecture/GFPGAN/gfpganv1_clean_arch.py", line 335, in forward
    feat = self.conv_body_down[i](feat)
  File "/Users/imt/Library/Application Support/chaiNNer/python/python/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/Applications/chaiNNer.app/Contents/Resources/src/nodes/utils/architecture/GFPGAN/gfpganv1_clean_arch.py", line 162, in forward
    out = F.interpolate(
  File "/Users/imt/Library/Application Support/chaiNNer/python/python/lib/python3.9/site-packages/torch/nn/functional.py", line 3731, in interpolate
    return torch._C._nn.upsample_bilinear2d(input, output_size, align_corners, scale_factors)
RuntimeError: Input and output sizes should be greater than 0, but got input (H: 1, W: 1) output (H: 0, W: 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Applications/chaiNNer.app/Contents/Resources/src/run.py", line 243, in run
    await executor.run()
  File "/Applications/chaiNNer.app/Contents/Resources/src/process.py", line 487, in run
    await self.__process_nodes(self.__get_output_nodes())
  File "/Applications/chaiNNer.app/Contents/Resources/src/process.py", line 475, in __process_nodes
    await self.process(output_node)
  File "/Applications/chaiNNer.app/Contents/Resources/src/process.py", line 282, in process
    return await self.__process(node)
  File "/Applications/chaiNNer.app/Contents/Resources/src/process.py", line 309, in __process
    processed_input = await self.process(node_input.id)
  File "/Applications/chaiNNer.app/Contents/Resources/src/process.py", line 282, in process
    return await self.__process(node)
  File "/Applications/chaiNNer.app/Contents/Resources/src/process.py", line 365, in __process
    raise NodeExecutionError(node, str(e), input_dict) from e
process.NodeExecutionError: Input and output sizes should be greater than 0, but got input (H: 1, W: 1) output (H: 0, W: 0)
```

--- a/backend/src/nodes/nodes/pytorch/interpolate_models.py
+++ b/backend/src/nodes/nodes/pytorch/interpolate_models.py
@@ -74,11 +74,17 @@ class InterpolateNode(NodeBase):
             return False
         interp_50 = self.perform_interp(model_a, model_b, 50)
         model = load_state_dict(interp_50).cpu()
-        fake_img = np.ones((3, 3, model.in_nc), dtype=np.float32)
+        size = (
+            model.min_size_restriction if model.min_size_restriction is not None else 3
+        )
+        assert isinstance(size, int), "min_size_restriction must be an int"
+        fake_img = np.ones((size, size, model.in_nc), dtype=np.float32)
         del interp_50
         with torch.no_grad():
             img_tensor = np2tensor(fake_img, change_range=True).cpu()
             t_out = model(img_tensor)
+            if isinstance(t_out, tuple):
+                t_out = t_out[0]
             result = tensor2np(t_out.detach(), change_range=False, imtype=np.float32)
         del model, img_tensor, t_out, fake_img
         mean_color = np.mean(result)

--- a/backend/src/nodes/utils/architecture/GFPGAN/gfpgan_bilinear_arch.py
+++ b/backend/src/nodes/utils/architecture/GFPGAN/gfpgan_bilinear_arch.py
@@ -187,6 +187,7 @@ class GFPGANBilinear(nn.Module):
         self.input_is_latent = input_is_latent
         self.different_w = different_w
         self.num_style_feat = num_style_feat
+        self.min_size_restriction = 512
 
         unet_narrow = narrow * 0.5  # by default, use a half of input channels
         channels = {

--- a/backend/src/nodes/utils/architecture/GFPGAN/gfpganv1_clean_arch.py
+++ b/backend/src/nodes/utils/architecture/GFPGAN/gfpganv1_clean_arch.py
@@ -215,6 +215,7 @@ class GFPGANv1Clean(nn.Module):
 
         self.supports_fp16 = False
         self.supports_bf16 = True
+        self.min_size_restriction = 512
 
         self.input_is_latent = input_is_latent
         self.different_w = different_w

--- a/backend/src/nodes/utils/architecture/GFPGAN/restoreformer_arch.py
+++ b/backend/src/nodes/utils/architecture/GFPGAN/restoreformer_arch.py
@@ -705,6 +705,7 @@ class RestoreFormer(nn.Module):
 
         self.supports_fp16 = False
         self.supports_bf16 = True
+        self.min_size_restriction = 16
 
         self.encoder = MultiHeadEncoder(
             ch=ch,

--- a/backend/src/nodes/utils/architecture/HAT.py
+++ b/backend/src/nodes/utils/architecture/HAT.py
@@ -914,6 +914,7 @@ class HAT(nn.Module):
         self.sub_type = "SR"
         self.supports_fp16 = False
         self.support_bf16 = True
+        self.min_size_restriction = 16
 
         state_keys = list(state_dict.keys())
 

--- a/backend/src/nodes/utils/architecture/RRDB.py
+++ b/backend/src/nodes/utils/architecture/RRDB.py
@@ -81,6 +81,7 @@ class RRDBNet(nn.Module):
 
         self.supports_fp16 = True
         self.supports_bfp16 = True
+        self.min_size_restriction = None
 
         # Detect if pixelunshuffle was used (Real-ESRGAN)
         if self.in_nc in (self.out_nc * 4, self.out_nc * 16) and self.out_nc in (

--- a/backend/src/nodes/utils/architecture/SPSR.py
+++ b/backend/src/nodes/utils/architecture/SPSR.py
@@ -65,6 +65,7 @@ class SPSRNet(nn.Module):
 
         self.supports_fp16 = True
         self.supports_bfp16 = True
+        self.min_size_restriction = None
 
         n_upscale = int(math.log(self.scale, 2))
         if self.scale == 3:

--- a/backend/src/nodes/utils/architecture/SRVGG.py
+++ b/backend/src/nodes/utils/architecture/SRVGG.py
@@ -47,6 +47,7 @@ class SRVGGNetCompact(nn.Module):
 
         self.supports_fp16 = True
         self.supports_bfp16 = True
+        self.min_size_restriction = None
 
         self.body = nn.ModuleList()
         # the first conv

--- a/backend/src/nodes/utils/architecture/SwiftSRGAN.py
+++ b/backend/src/nodes/utils/architecture/SwiftSRGAN.py
@@ -125,6 +125,7 @@ class Generator(nn.Module):
 
         self.supports_fp16 = True
         self.supports_bfp16 = True
+        self.min_size_restriction = None
 
         self.initial = ConvBlock(
             in_channels, num_channels, kernel_size=9, stride=1, padding=4, use_bn=False

--- a/backend/src/nodes/utils/architecture/Swin2SR.py
+++ b/backend/src/nodes/utils/architecture/Swin2SR.py
@@ -1043,6 +1043,7 @@ class Swin2SR(nn.Module):
 
         self.supports_fp16 = False  # Too much weirdness to support this at the moment
         self.supports_bfp16 = True
+        self.min_size_restriction = 16
 
         ## END AUTO DETECTION
 

--- a/backend/src/nodes/utils/architecture/SwinIR.py
+++ b/backend/src/nodes/utils/architecture/SwinIR.py
@@ -977,6 +977,7 @@ class SwinIR(nn.Module):
 
         self.supports_fp16 = False  # Too much weirdness to support this at the moment
         self.supports_bfp16 = True
+        self.min_size_restriction = 16
 
         self.img_range = img_range
         if in_chans == 3:

